src/elisp/treemacs-mode.el

```diff
@@ -226,10 +226,15 @@ Will be set by `treemacs--post-command'.")
               ((memq 'moody-mode-line-buffer-identification
                      (default-value 'mode-line-format))
                '(:eval (moody-tab " Treemacs " 10 'down)))
-              ((and (fboundp 'doom-modeline)
-                    (fboundp 'doom-modeline-def-modeline))
-               (doom-modeline-def-modeline 'treemacs '(bar " " major-mode) '())
-               (doom-modeline 'treemacs))
+              ((fboundp 'doom-modeline)
+	       (eval-and-compile
+		 (require 'doom-modeline)
```

Comment:
Should use `with-eval-after-load` instead of `require` since `doom-modeline` is not a dependency.

Reply:
```
              ((and (fboundp 'doom-modeline)
		    (fboundp 'doom-modeline-def-segment)
		    (fboundp 'doom-modeline-face)
		    (fboundp 'doom-modeline-def-modeline))
	       (with-eval-after-load 'doom-modeline)
	       (doom-modeline-def-segment treemacs-workspace-name
		 "Display treemacs workspace name."
		 (propertize (format " %s " (treemacs-workspace->name (treemacs-current-workspace)))
			     'face (doom-modeline-face 'doom-modeline-buffer-minor-mode)))
	       (doom-modeline-def-modeline 'treemacs '(bar " " major-mode) '(treemacs-workspace-name))
	       (doom-modeline 'treemacs))
```

This triggers `reference to free variable 'treemacs-workspace-name'` again...

Reply:
```emacs-lisp
((fboundp 'doom-modeline)
 (with-eval-after-load 'doom-modeline
   (doom-modeline-def-segment treemacs
     "Display treemacs."
     (propertize (format " %s " (treemacs-workspace->name (treemacs-current-workspace)))
                 'face (doom-modeline-face 'doom-modeline-buffer-minor-mode)))
   (doom-modeline-def-modeline 'treemacs '(bar " " major-mode) '(treemacs))
   (doom-modeline 'treemacs))
 )
```
Got this error

```
Debugger entered--Lisp error: (void-variable treemacs)
  #<subr F616e6f6e796d6f75732d6c616d626461_anonymous_lambda_8>()
  eval-after-load(doom-modeline #<subr F616e6f6e796d6f75732d6c616d626461_anonymous_lambda_8>)
  treemacs--setup-mode-line()
  #f(compiled-function () (interactive nil) #<bytecode -0x1b2d643fdaa15b7>)()
  treemacs--mode-check-advice(#f(compiled-function () (interactive nil) #<bytecode -0x1b2d643fdaa15b7>))
  apply(treemacs--mode-check-advice #f(compiled-function () (interactive nil) #<bytecode -0x1b2d643fdaa15b7>) nil)
  treemacs-mode()
```

I think we need `eval-and-compile` because we need expand `doom-modeline-def-segment` macro.

Reply:
If `doom-modeline` is byte-compiled and loaded before treemacs, no such errors occur IMO.